libheif/plugins/heif_encoder_aom.cc

```diff
@@ -324,6 +323,7 @@ static void aom_init_parameters()
   p->has_default = true;
   d[i++] = p++;
 
+  assert(i < MAX_NPARAMETERS + 1);
```

Comment:
I used `MAX_NPARAMETERS + 1` in this assertion to match the corresponding array definition:

```
static const struct heif_encoder_parameter* aom_encoder_parameter_ptrs[MAX_NPARAMETERS + 1];
```

We could use the equivalent `assert(i <= MAX_NPARAMETERS);` instead.